extensions/ql-vscode/src/query-history.ts

```diff
@@ -50,6 +50,15 @@ const SHOW_QUERY_TEXT_QUICK_EVAL_MSG = `\
  */
 const FAILED_QUERY_HISTORY_ITEM_ICON = 'media/red-x.svg';
 
+export async function updateTreeItemContextValue(element: CompletedQuery): Promise<void> {
```

Comment:
Would this make more sense as a member function on `CompletedQuery`?

Reply:
I think it makes sense as a member function on `QueryHistoryManager`, because of my attempting to decouple `CompletedQuery` from having to know too much about the whole `QueryHistoryManager` and the associated tree data provider. Already I don't love having to put the tree item itself as a field in `CompletedQuery`, but this is the simplest thing that I can see works.